feat: Add vsphere-supervisor builder (#220)
hashicorp/packer-plugin-vsphere: README.md

* `vsphere-clone` - This builder clones an existing virtual machine template, modifies the template, and then saves it as a new image. The builder uses the vSphere API to build images on a vSphere cluster or ESXi host by connecting to a vCenter Server instance.

> * `vsphere-supervisor` - This builder deploys a vm to a vSphere Supervisor cluster by using the VM-Service API.

Reviewer: ```suggestion
* `vsphere-supervisor` - This builder creates a virtual machine on  a vSphere Supervisor cluster by using the VM-Service API.
```

Reviewer: This may be more accurate. May need an update after adding the publish capabilities.
